src/components/error-dart.js

```diff
@@ -1,11 +1,71 @@
 import React from 'react'
+import axios from 'axios'
 import { Header, Button } from 'semantic-ui-react'
 
-import { getTheme, urlErrorAssets } from 'formula_one'
+import { getTheme, urlErrorAssets, urlWhoAmI } from 'formula_one'
 
 import blocks from '../css/error-dart.css'
+import { element } from 'prop-types'
 
 export default class ErrorDart extends React.PureComponent {
+  state = {
+    isFacultyOrGuest: false
+  }
+
+  isFacultyOrGuest = (roles) => {
+    let isFacultyOrGuest = false
+    roles.forEach(element => {
+      if (element.role === 'FacultyMember' || element.role === 'Guest') {
+        isFacultyOrGuest = true
+      }
+    })
+    return isFacultyOrGuest
+  }
+
+  fetchUserRoles = () => {
```

Comment:
Actually this API should not be called there, also I believe this API is already called and stored in a reduc state. So try if we can extract from there.

Reply:
Actually, it was called when the AppComponent got rendered which does not necessarily render the ErrorDart or ErrorRabbit component. Now, I have called this API endpoint in the NoMatch component which renders either ErrorDart or ErrorRabbitc component.